cmd/oras/internal/option/remote.go

```diff
@@ -322,3 +326,32 @@ func (opts *Remote) isPlainHttp(registry string) bool {
 	}
 	return plainHTTP
 }
+
+// Handle handles error during cmd execution.
+func (opts *Remote) Handle(err error, cmd *cobra.Command) (oerrors.Processor, error) {
+	// handle not found error from registry
+	if errors.Is(err, errdef.ErrNotFound) {
+		cmd.SetErrPrefix(oerrors.RegistryErrorPrefix)
+		return opts, err
+	}
+
+	// handle error response
+	var errResp *errcode.ErrorResponse
+	if errors.As(err, &errResp) {
+		cmd.SetErrPrefix(oerrors.RegistryErrorPrefix)
+		return opts, err
+	}
+	return nil, err
+}
+
+// Process processes error into oerrors.Error.
+func (opts *Remote) Process(err error, _ string) *oerrors.Error {
+	ret := oerrors.Error{
+		Err: err,
+	}
+	var errResp *errcode.ErrorResponse
+	if errors.As(err, &errResp) {
+		ret.Err = oerrors.GetInner(err, errResp)
```

Comment:
This pattern seems can be further refactored.

Reply:
Could you explain more detail on how to do that?